src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -619,12 +621,22 @@ namespace smt::noodler {
         conjuncts.push_back(conv_form_with_precision.first);
         precision = conv_form_with_precision.second;
 
+        bool has_lia_reducible_predicates = (!solution.length_sensitive_vars.empty() || !this->not_contains.get_predicates().empty() || !this->disequations.get_predicates().empty());
```

Comment:
This condition is negation of the condition on the line 598, so `has_lia_reducible_predicates` should always be `true`. Is this correct?

Reply:
Yes, your conclusion is correct. Originally the conditions were different, but during the development they eventually converged together. I have removed the boolean var and the subsequent if.